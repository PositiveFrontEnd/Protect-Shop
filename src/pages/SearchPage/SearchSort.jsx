import React, { useState } from 'react';
import ArrowDown from './SearchSvg/arrowDown.svg?react';
import ArrowUp from './SearchSvg/arrowUp.svg?react';
import PropTypes from "prop-types";

const Sort = ({ handleSort, sort }) => {
    const [isOpenSort, setIsOpenSort] = useState(false);
    const actionSort = () => {
        setIsOpenSort(!isOpenSort)
    };
    return (
        <div>
            <button className={isOpenSort ? 'sort-open' : 'sort'}
                onClick={actionSort}>sort
                {
                    isOpenSort ? <ArrowUp /> : <ArrowDown />
                }
            </button>
            {isOpenSort && (
                <div className='sort__dropdown'>
                    <label><input type='radio'
                        value='lowToHigh'
                        checked={sort === 'lowToHigh'}
                        onChange={handleSort} />
                        Price low to high</label>
                    <label><input type='radio'
                        value='highToLow'
                        checked={sort === 'highToLow'}
                        onChange={handleSort} />
                        Price high to low</label>
                </div>
            )}
        </div>
    );
};
Sort.propTypes = {
    handleSort: PropTypes.func,
    sort: PropTypes.string
}
export default Sort;