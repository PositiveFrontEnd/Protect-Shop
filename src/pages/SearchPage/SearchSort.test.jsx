import React from "react";
import { render } from "@testing-library/react";
import SearchSort from "./SearchSort";
import { BrowserRouter } from "react-router-dom";
describe("Тестування компонента  SearchSort", () => {
    test("Знімок", () => {

        const data = {
            handleSort: () => console.log(),
            sort: 'highToLow'
        }

        const shopsPage = render(
            <BrowserRouter>
                <SearchSort data={data} />
            </BrowserRouter>
        );
        expect(shopsPage).toMatchSnapshot();
    });
});
