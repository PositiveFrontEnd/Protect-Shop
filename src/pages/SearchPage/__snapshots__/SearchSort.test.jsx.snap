// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  SearchSort > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          class="sort"
        >
          sort
          <svg
            fill="none"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.13828 6.54375L5.10391 5.57812L8.00078 8.475L10.8977 5.57812L11.8633 6.54375L8.00078 10.4062L4.13828 6.54375Z"
              fill="#191919"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        class="sort"
      >
        sort
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.13828 6.54375L5.10391 5.57812L8.00078 8.475L10.8977 5.57812L11.8633 6.54375L8.00078 10.4062L4.13828 6.54375Z"
            fill="#191919"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
